onfido-java/src/main/java/com/onfido/ExtractionManager.java

```diff
@@ -24,12 +23,12 @@ protected ExtractionManager(Config config, OkHttpClient client) {
     /**
      * Initiates an extraction for a document.
      *
-     * @param request the request body
+     * @param document_id the unique identifier of the uploaded document to run extraction on
      * @return the Extraction
      * @throws OnfidoException the onfido exception
      */
-    public Extraction create(Extraction.Request request) throws OnfidoException {
-        return extractionParser.parse(post("", requestFormatter.toJson(request)));
+    public Extraction perform(String document_id) throws OnfidoException {
+        return extractionParser.parse(post("", String.format("{ document_id: %s }", document_id)));
```

Comment:
I forgot this is sent as JSON. Kind of annoying. Let's do it this way for now, but you'll need to do some quoting:

`String.format("{ \"document_id\": \"%s\" }"`

The test I recommended should catch issues though.

Reply:
Thanks for catching, updated